### PR TITLE
[New Rule] Potential Widespread Malware Infection

### PR DESCRIPTION
**Pull request by w0rk3r**
## Summary

v2 of https://github.com/elastic/detection-rules/pull/2402

This rule uses alert data to determine when a malware signature is triggered in multiple hosts. Analysts can use this to prioritize triage and response, as this can potentially indicate a widespread malware infection.

image

### CODE REVIEW

File: rules/cross-platform/execution_potential_widespread_malware_infection.toml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    "Use Case: Threat Detection",
+    "Tactic: Execution",
+    "Rule Type: Higher-Order Rule"
+]
+timestamp_override = "event.ingested"
+type = "esql"
+
+query = '''
+from logs-endpoint.alerts-*
+| where event.code == "memory_signature"
```
[the remark this answers is not in the document]
Comment (w0rk3r) [reply rules/cross-platform/execution_potential_widespread_malware_infection.toml:36]: Awesome suggestion

**Review by Samirbous — APPROVED** [review]

File: rules/cross-platform/execution_potential_widespread_malware_infection.toml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+[rule]
+author = ["Elastic"]
+description = """
+This rule uses alert data to determine when a malware signature is triggered in multiple hosts. Analysts can use this
+to prioritize triage and response, as this can potentially indicate a widespread malware infection.
+"""
+from = "now-9m"
+language = "esql"
+license = "Elastic License v2"
+name = "Potential Widespread Malware Infection"
```
Comment (brokensound77) [inline rules/cross-platform/execution_potential_widespread_malware_infection.toml:17]:
```suggestion
name = "Potential Widespread Malware Infection Across Multiple Hosts"
```

File: rules/cross-platform/execution_potential_widespread_malware_infection.toml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+]
+risk_score = 73
+rule_id = "28371aa1-14ed-46cf-ab5b-2fc7d1942278"
+severity = "high"
+tags = [
+    "Domain: Endpoint",
+    "Data Source: Elastic Defend",
+    "Use Case: Threat Detection",
+    "Tactic: Execution",
+    "Rule Type: Higher-Order Rule"
```
Comment (brokensound77) [inline rules/cross-platform/execution_potential_widespread_malware_infection.toml:29]: is it?

File: rules/cross-platform/execution_potential_widespread_malware_infection.toml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+]
+timestamp_override = "event.ingested"
+type = "esql"
+
+query = '''
+from logs-endpoint.alerts-*
+| where event.code in ("malicious_file", "memory_signature", "shellcode_thread") and rule.name is not null
+| stats hosts = count_distinct(host.id) by rule.name,event.code
```
Comment (brokensound77) [inline rules/cross-platform/execution_potential_widespread_malware_infection.toml:37]:
```suggestion
| stats hosts = count_distinct(host.id) by rule.name, event.code
```

**Review by brokensound77 — APPROVED** [review]
few suggestions then LGTM

File: rules/cross-platform/execution_potential_widespread_malware_infection.toml (hunk 3 above)
Comment (w0rk3r) [reply rules/cross-platform/execution_potential_widespread_malware_infection.toml:29]: I think so, it uses alerts as the data source

**Pull request merged**